app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/dialogs/EnvironmentOptionsWidget.java

```diff
@@ -0,0 +1,202 @@
+import android.view.View;
+import android.widget.ScrollView;
+
+import org.mozilla.vrbrowser.R;
+import org.mozilla.vrbrowser.audio.AudioEngine;
+import org.mozilla.vrbrowser.browser.SessionStore;
+import org.mozilla.vrbrowser.browser.SettingsStore;
+import org.mozilla.vrbrowser.ui.views.UIButton;
+import org.mozilla.vrbrowser.ui.views.settings.ButtonSetting;
+import org.mozilla.vrbrowser.ui.views.settings.RadioGroupSetting;
+import org.mozilla.vrbrowser.ui.views.settings.SwitchSetting;
+import org.mozilla.vrbrowser.ui.widgets.UIWidget;
+import org.mozilla.vrbrowser.ui.widgets.WidgetManagerDelegate;
+import org.mozilla.vrbrowser.ui.widgets.WidgetPlacement;
+import org.mozilla.vrbrowser.utils.LocaleUtils;
+
+public class EnvironmentOptionsWidget extends UIWidget implements
+        WidgetManagerDelegate.WorldClickListener,
+        WidgetManagerDelegate.FocusChangeListener {
+
+    private AudioEngine mAudio;
+    private UIButton mBackButton;
+
+    private SwitchSetting mEnvOverrideSwitch;
+    private RadioGroupSetting mEnvironmentsRadio;
+    private int mRestartDialogHandle = -1;
+
+    private ButtonSetting mResetButton;
+
+    private ScrollView mScrollbar;
+
+    public EnvironmentOptionsWidget(Context aContext) {
+        super(aContext);
+        initialize(aContext);
+    }
+
+    public EnvironmentOptionsWidget(Context aContext, AttributeSet aAttrs) {
+        super(aContext, aAttrs);
+        initialize(aContext);
+    }
+
+    public EnvironmentOptionsWidget(Context aContext, AttributeSet aAttrs, int aDefStyle) {
+        super(aContext, aAttrs, aDefStyle);
+        initialize(aContext);
+    }
+
+    private void initialize(Context aContext) {
+        inflate(aContext, R.layout.options_environment, this);
+
+        mAudio = AudioEngine.fromContext(aContext);
+
+        mWidgetManager.addFocusChangeListener(this);
+        mWidgetManager.addWorldClickListener(this);
+
+        mBackButton = findViewById(R.id.backButton);
+        mBackButton.setOnClickListener(view -> {
+            if (mAudio != null) {
+                mAudio.playSound(AudioEngine.Sound.CLICK);
+            }
+
+            onDismiss();
+        });
+
+        String env = SettingsStore.getInstance(getContext()).getEnvironment();
+        mEnvironmentsRadio = findViewById(R.id.environmentRadio);
+        mEnvironmentsRadio.setOnCheckedChangeListener(mEnvsListener);
+        setEnv(mEnvironmentsRadio.getIdForValue(env), false);
+
+        mEnvOverrideSwitch = findViewById(R.id.envOverrideSwitch);
+        mEnvOverrideSwitch.setOnCheckedChangeListener(mEnvOverrideListener);
+        setEnvOverride(SettingsStore.getInstance(getContext()).isEnvironmentOverrideEnabled());
+        mEnvOverrideSwitch.setHelpDelegate(() -> {
+            if (mAudio != null) {
+                mAudio.playSound(AudioEngine.Sound.CLICK);
+            }
+            SessionStore.get().loadUri(getContext().getString(R.string.environment_override_help_url));
```

Comment:
this is helpful. thanks for adding :+1:

<hr>

P.S. UX nit request: to make it easier for developers who will likely be reading documentation from their desktop PC screens, it might be nice to have @philip-lamb submit a bit.ly request for a short URL (à la https://mzl.la/fxr or https://mzl.la/reality), and the tiny URL could possibly be shown in a tooltip upon hover.

![image](https://user-images.githubusercontent.com/203725/54202814-0fa49580-448e-11e9-8a6a-efee1ee4ce81.png)